src/lib/storage/lz4_segment/lz4_segment_iterable.hpp

```diff
@@ -19,18 +19,15 @@ class LZ4SegmentIterable : public PointAccessibleSegmentIterable<LZ4SegmentItera
   template <typename Functor>
   void _on_with_iterators(const Functor& functor) const {
     using ValueIterator = typename std::vector<T>::const_iterator;
+    using NullValueIterator = typename pmr_vector<bool>::const_iterator;
```

Comment:
For the current TPC-H, it's not much. As soon as a query on a sorted segment is executed, it's a lot (can be factors, depending on input size, selectivity etc.). For a reason I do not understand, `AnySegmentIterables` do not support binary searches (they should be random accessible though).

With the change to the segment accessors (I actually don't dare to say it will end up in the master, as they main advantage is for caching iterators and that's only LZ4), it makes again more sense as iterables are reused much more often.

So in short: currently not much, later maybe more. :)
So always erasing them is still the correct decision imo.